app/scripts/controllers/network/network-controller.test.js

```diff
@@ -4639,6 +4641,110 @@ describe('NetworkController', () => {
       });
     });
   });
+  describe('upsertNetworkConfiguration', () => {
+    it('should add a network configuration', async () => {
+      await withController(
+        {
+          state: {
+            networkConfigurations: {},
+          },
+        },
+        async ({ controller }) => {
+          const rpcUrlNetwork = {
+            chainId: undefined,
+            chainName: 'RPC',
+            rpcPrefs: undefined,
+            rpcUrl: 'rpc_url',
+            ticker: 'RPC',
+          };
+
+          expect(controller.networkConfigurations.getState()).toStrictEqual({});
+
+          controller.upsertNetworkConfiguration(rpcUrlNetwork);
+
+          expect(
+            Object.values(controller.networkConfigurations.getState()),
+          ).toStrictEqual(expect.arrayContaining([rpcUrlNetwork]));
+        },
+      );
+    });
+
+    it('should update a network configuration when the configuration being added has an rpcURL that matches an existing configuration', async () => {
```

Comment:
I wonder if this could be reworded slightly as well, as "the configuration being is added" isn't being added. Perhaps something like:

``` suggestion
    it('should use the given configuration to update the existing network configuration that matches its rpcUrl', async () => {
```

Reply:
changed [here](https://github.com/MetaMask/metamask-extension/pull/17421/commits/febce08fe121e0564df73d59772a3a5fec8fd855)